04.class/lib/memo_app.js

```diff
@@ -0,0 +1,86 @@
+import { Memo } from "./memo.js";
+import { View } from "./view.js";
+
+export class MemoApp {
+  #view = null;
+
+  constructor() {
+    this.#view = new View();
+  }
+
+  async run(params) {
+    await Memo.createTable();
+
+    if (params.l) {
+      await this.#printMemos();
+    } else if (params.r) {
+      await this.#printMemo();
+    } else if (params.d) {
+      await this.#deleteMemo();
+    } else {
+      await this.#createMemo();
+    }
+
+    await Memo.close();
+  }
+
+  async #printMemos() {
+    const memos = await Memo.all();
+    if (memos.length === 0) {
+      this.#view.printNoMemos();
+    } else {
+      this.#view.printMemos(memos);
+    }
+  }
+
+  async #printMemo() {
+    const selectedTitle = await this.#displayMemoSelectionPrompt("see");
+    if (!selectedTitle) {
+      this.#view.printNoMemos();
+      return;
+    }
+
+    const memo = await Memo.findByTitle(selectedTitle);
+    this.#view.printMemo(memo);
+  }
+
+  async #deleteMemo() {
+    const selectedTitle = await this.#displayMemoSelectionPrompt("delete");
+    if (!selectedTitle) {
+      this.#view.printNoMemos();
+      return;
+    }
+
+    await Memo.deleteByTitle(selectedTitle);
```

Comment:
```suggestion
    return await doHogeHoge("delete", (selectedTitle) => {
      // ...
    });
```

みたいなメソッドを作るとprintMemoと処理を共通化できるかも？